@xen-orchestra/rest-api/src/rest-api/rest-api.type.mts

```diff
@@ -43,10 +45,10 @@ export type XoApp = {
     opts?: { bypassOtp?: boolean }
   ) => Promise<{ bypassOtp: boolean; expiration: number; user: XoUser }>
   getAllXenServers(): Promise<XoServer[]>
-  getObject: <T extends XapiXoRecord>(id: T['id'], type: T['type']) => T
+  getObject: <T extends XapiXoRecord>(id: T['id'], type?: T['type']) => T
```

Comment:
I think it should stay mandatory, even if some subclasses don't uses it

Reply:
FYI, the `type` parameter in the `getObject` is really optional.
See https://github.com/vatesfr/xen-orchestra/blob/master/packages/xo-server/src/xo.mjs#L80